services/frontend-service/pkg/auth/auth.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"context"
+	"google.golang.org/grpc/metadata"
+	"net/http"
+)
+
+type ctxMarker struct{}
+
+var (
+	ctxMarkerKey = &ctxMarker{}
+	defUser      = &User{
+		Email:    "local@freiheit.com",
+		Username: "local@freiheit.com",
+	}
+)
+
+// Extract takes the User from middleware.
+// It always returns a User
+func Extract(ctx context.Context) *User {
+	md, _ := metadata.FromIncomingContext(ctx)
+	u := defUser
```

Comment:
what does "def" stand for? I guess "default"? Please then call it "defaultUser"

Reply:
Done!